leshan-core/src/main/java/org/eclipse/leshan/core/node/LwM2mSingleResource.java

```diff
@@ -116,6 +117,11 @@ public static LwM2mSingleResource newResource(int id, Object value, Type type) {
                 throw new LwM2mNodeException(
                         String.format(doesNotMatchMessage, value.getClass().getSimpleName(), type));
             break;
+        case CORELINK:
+            if (!(value instanceof Link[]))
+                throw new LwM2mNodeException(
+                        String.format(doesNotMatchMessage, value.getClass().getSimpleName(), type));
+            break;
```

Comment:
I guess a new static constructor should be added for corelink in this class ?
And probably this should be also be implemented in `LwM2mResourceInstance` and `LwM2mMultipleResource`

Reply:
done